R/tools.R

```diff
@@ -26,6 +26,52 @@ lint_dir <- function(path) {
   lints
 }
 
+lint_file <- function(path) {
+  if (interactive()) {
+    message(cli::format_inline("Linting {.file {path}}"))
+  }
+  lints <- lintr::lint(path)
+  # Use the actual path provided by the user (typically relative) to make results more readable.
+  # (`lintr::lint()` normalizes paths, e.g. `app/main.R` becomes a full absolute path.)
+  for (i in seq_along(lints)) {
+    lints[[i]]$filename <- path
+  }
+  lints
+}
+
+lint_path <- function(path) {
+  # Check if path is a directory or a file such that nothing happens when the path is invalid.
+  if (fs::is_dir(path)) {
+    lint_dir(path)
+  } else if (fs::is_file(path)) {
+    lint_file(path)
+  }
+}
+
+check_paths <- function(paths) {
```

Comment:
Nice! I think a message coming from us will be more user-friendly then what `lintr` prints :+1:

I see some danger in this interface though. A user might make a typo in the file name and not notice that it was skipped:
```
This path does not exist: misatke.
Linting app...
✔ No style errors found.
```
Instead let's require that all paths provided by the user are valid and fail immediately if that's not the case.